src/shogun/base/SGObject.cpp

```diff
@@ -254,6 +254,7 @@ void CSGObject::update_parameter_hash()
 	uint32_t carry=0;
 	uint32_t length=0;
 
+	m_hash=0;
```

Comment:
please send a seperate PR for this fix, and add unit tests that make sure the rationale is explicit